src/common.rs

```diff
@@ -150,6 +150,11 @@ impl Errors {
     }
 }
 
+/// Trait used to bundle up resources/components for easy registration with the `World`.
+pub trait WorldResourceRegistrator {
+    fn register_with_world(self, world: &mut World);
```

Comment:
Register implies a restriction to components, could you rename it to `add*`, please?